eclipse/src/saros/SarosEclipseContextFactory.java

```diff
@@ -131,15 +130,30 @@ public void createComponents(MutablePicoContainer container) {
         BindKey.bindKey(String.class, IContextKeyBindings.PlatformVersion.class),
         Platform.getBundle("org.eclipse.core.runtime").getVersion().toString());
 
-    // for core logic and extended Eclipse session components
+    final org.eclipse.jface.preference.IPreferenceStore instanceScopePreferenceStore =
+        saros.getPreferenceStore();
+
+    container.addComponent(
+        IPreferenceStore.class, new EclipsePreferenceStoreAdapter(instanceScopePreferenceStore));
+
+    /* FIXME the class states that it should be the global preferences, however it does not have method to write preferences and so it is fine for now to use the instance scope */
+    //    container.addComponent(
```

Comment:
What is this commented out code doing here? Is this the logic that is supposed to be added once global preferences are possible?

Reply:
It would change the preference store to a global one, i.e shared across all Eclipse Workspaces.

I did not changed it yet because we should look what settings should be really global. And I do not really find it a good idea to talking about global stuff in the core Preferences. It should be up to the caller do decide what scope these preferences have.